keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+# Azure Key Vault Secret client library for Java
+Azure Key Vault is a tool for securely storing and accessing secrets. A secret is anything that you want to tightly control access to, such as API keys or passwords. A vault is logical group of secrets.
+Secret client library allows you to securely store and tightly control access to tokens, passwords, API keys, and other secrets. THe library offers operations to create, retrieve, update, delete, purge, backup, restore and and list the secrets.
+
+Use the secret client library to create and manage secrets.
+
+[Source code][source_code] | [Package (Maven)][package] | [API reference documentation][api_documentation] | [Product documentation][azkeyvault_docs]
+
+## Getting started
+
+### Prerequisites
+
+- [Java Development Kit (JDK)][jdk] with version 8 or above
+- [Azure Subscription][azure_subscription]
+- [Azure KeyVault][azure_keyvault]
+
+### Adding the package to your product
+
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-keyvault-secrets</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+### Create a Secret in the Azure Key Vault.
+
+To create an Azure Key Vault Store you can use the Azure Portal or [Azure CLI][azure_cli].
+
+Create the Azure Key Vault:
+```Powershell
+az keyvault create --name <keyvault-name> --resource-group <resource-group-name> --location eastus
+```
+
+After that, create the secret in the Azure Key Vault:
+```Powershell
+az keyvault secret set --vault-name <keyvault-name> --name "<secret-name>" --value "<secret-value>"
+```
+
+### Authenticate the client
+
+Applications that use a key vault must authenticate by using a token from Azure Active Directory. The owner of the application must register it in Azure Active Directory first. At the end of registration, the application owner gets the following values:
+
+* An Application ID (also known as the AAD Client ID or appID)
+* An authentication key (also known as the shared secret).
+
+The application must present both these values to Azure Active Directory, to get a token. 
+
+#### Create/Get Credentials
+
+Create an application in the Azure Active Directory.
+```Powershell
+az ad sp create-for-rbac -n <application-name> --password <application-password> --skip-assignment
+# If you don't specify a password, one will be created for you.
+```
+
+To authorize the same application to perform secret operarions in your vault, type the following command:
+```Powershell
+az keyvault set-policy --name <keyvault-name> --spn 8f8c4bbd-485b-45fd-98f7-ec6300b7b4ed --secret-permissions <secret-permissions>
```

Comment:
I'd replace the guid with a placeholder like `<service principal ID>`. Should this be the ID of the application or the service principal? What's the literal name used in the output of `create-for-rbac`? 

And what should `--secret-permissions` be?